Feature/awf/upstream commits 20170116

CloudNiner opened the pull request:
## Overview

Updates our subproject with all relevant upstream commits.


## Notes

Manually reverts the SRID change to 3857 that occurred in e1c7d31100f73e68d29bcd72c8f31e7e40d19c8d within the merge commit.

860417ab268069857a6592e34b9f30d4f7d126fc was essentially a no-op as we updated all the 'cambridge' references earlier.


## Testing

The analysis should continue to successfully run with any neighborhood input. Tested this PR with the Cambridge shapefile.

Closes:
 - azavea/pfb-network-connectivity#36
 - azavea/pfb-network-connectivity#37
 - azavea/pfb-network-connectivity#38

spencerrecneps committed:
fix incorrect ft/tf column reference

spencerrecneps committed:
prep destinations layer for streetlightdata

spencerrecneps committed:
add rules for better handling of lane data

spencerrecneps committed:
add cross lanes and rework lane assignment

spencerrecneps committed:
refine handling of cross lanes

spencerrecneps committed:
change references to cambridge into neighborhood

spencerrecneps committed:
change srid to 3857, update spatial calculations to meters from feet

Commit:
Merge remote-tracking branch 'upstream/master' into feature/awf/upstream-commits-20170116

- Reverts upstream commit e1c7d31100f73e68d29bcd72c8f31e7e40d19c8d,
  which updated the analysis to 3857. See
  https://github.com/azavea/pfb-network-connectivity/issues/39
  for why we're not using this.

sharph commented on connectivity/cycles.py:
```diff
@@ -2,7 +2,7 @@
 import networkx as nx
 
 zipCode = '02138'
```
Do you know the significance of this `zipCode`?

CloudNiner replied:
Oh good catch. It's the Cambridge MA zip code. I don't believe either of these scripts are in active use ATM, but I'll need to fixup the WHERE clause in your other comment anyways.

spencerrecneps replied:
Correct. Neither of the .py scripts are in use currently. Was a vestige of initial development as the tool was starting to take shape and before we settled on pg_routing.

sharph commented on streetlight/streetlight_destinations.sql:
```diff
@@ -0,0 +1,28 @@
+        0
+FROM    neighborhood_census_blocks blocks,
+        neighborhood_zip_codes zips
+WHERE   zips.zip_code = '02138'
```
and this one?

spencerrecneps replied:
This script isn't in use by the core tool at the moment but it should be updated. It will eventually be used with the StreetLightData option that is still being evaluated by PFB and likely to be OK'd.

Commit:
Add zipcode note in cycles.py

Commit:
Fixup streetlight_destinations.sql

- Continue to use SRID expected by StreetLightData
- Support change from static zipcode boundary to
  custom user provided boundary

CloudNiner commented:
Updated.

CloudNiner commented:
@spencerrecneps updated once again. Want to take quick look and make sure this is correct?

spencerrecneps commented:
@CloudNiner looks good to me. Thanks!
